### PR TITLE
Add RSVP feature for attending any event, including hangouts. Does no…

### PR DESCRIPTION
**Pull request by markd315**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Allows a user to register attendance intention for any event (including any hangouts)

* **What is the current behavior?** (You can also link to an open issue here)
No such

* **What is the new behavior (if this is a feature change)?**
new POST and DELETE endpoints supported by backend contract to link with the frontend feature by Juan

* **Other information**:
Postman collection in readme updated accordingly

### CODE REVIEW

File: src/domain/event/event.controller.ts [hunk 1]
```diff
@@ -1,19 +1,22 @@
...
-const eventService = new EventService();
+const profileRepository = new ProfileRepository();
 
 let _ = require("lodash");
 let moment = require("moment");
 let dateFormats = ["YYYY-MM-DD HH:mm", "YYYY-MM-DD"]; //prefer 24-hour time
 
 async function createEvent(req) {
-    const event = eventService.createEvent(req.body);
+    const event = EventService.createEvent(req.body);
```
Comment (audgster) [inline src/domain/event/event.controller.ts]: Why are the services static?

File: src/domain/event/event.controller.ts [hunk 2]
```diff
@@ -29,6 +32,41 @@ async function getAllEvents(req) {
...
+  let eventId = req.params.id;
+  let memberId = req.body.id;
+  let rsvpAnswer = req.body.attending;
+  let profile = await profileRepository.getProfile(memberId)
+
+  const member = EventService.createMember(profile, rsvpAnswer);
+  const validationResult = eventValidator.validateMember(member);
+
+  //then update the event to include the profile
+  return validationResult.isValid ? await eventRepository.addEventAttendee(eventId, member) : validationResult;
```
Comment (audgster) [inline src/domain/event/event.controller.ts]: May want to include try/catches around awaits should the promise fail

File: src/domain/event/event.repository.ts [hunk 3]
```diff
@@ -37,4 +37,67 @@ export default class EventRepository {
...
+      ReturnValues: "ALL_NEW"
+    };
+    const promise = this.dbClient.update(params).promise();
+    return await promise.then(data => {
+      return data.Attributes
+    })
+  }
+
+  async removeEventAttendee(eventId, attendeeIndex) {
+    const params = {
```
Comment (audgster) [inline src/domain/event/event.repository.ts]: Is there no delete in the client interface?

File: src/domain/event/event.routes.ts [hunk 4]
```diff
@@ -21,4 +21,40 @@ router.get('', function(req, res) {
   })
 });
 
+router.post('/:id/rsvp', function (req, res) {
```
Comment (audgster) [inline src/domain/event/event.routes.ts]: These aren't super REST-y but we can discuss.

File: src/domain/event/event.validator.ts [hunk 5]
```diff
@@ -12,4 +12,28 @@ export default class EventValidator {
...
+    this.validateMemberName(member.firstName, member.lastName, errors);
+
+    return {
+      errors: errors,
+      isValid: (errors.length == 0)
+    }
+  }
+
+  private validateMemberName(firstName, lastName, errors) {
+    if(firstName == undefined) {
```
Comment (audgster) [inline src/domain/event/event.validator.ts]: I feel like you can use the boolean `if(!firstName)` to do the same thing

File: src/domain/event/event.controller.ts (hunk 1 above)
Comment (markd315) [reply src/domain/event/event.controller.ts]: some others were, not for any good reason

File: src/domain/event/event.validator.ts (hunk 5 above)
Comment (markd315) [reply src/domain/event/event.validator.ts]: 👍

File: src/domain/event/event.controller.ts (hunk 1 above)
Comment (markd315) [reply src/domain/event/event.controller.ts]: 👍

File: src/domain/event/event.routes.ts (hunk 4 above)
Comment (markd315) [reply src/domain/event/event.routes.ts]: Let me know what you want to change it to in the future if you like, but without more details I'm gonna move on

File: src/domain/event/event.controller.ts (hunk 2 above)
Comment (markd315) [reply src/domain/event/event.controller.ts]: transparently choosing to ignore this comment, someone else can if needed 😆

**Comment by markd315** [comment]
Merged in other PR

**Pull request closed without merging**